### ISSUE TITLE
clrDgColType='number' broken

### ISSUE THREAD

**Issue by duzenko** [vmware/clarity#4567]
## Describe the bug

clrDgColType='number' is not doing any actual filtering

## How to reproduce

https://stackblitz.com/edit/clr-basic-datagrid-4zdewf?file=src%2Fapp%2Fbasic-datagrid.component.html

1.  Go to stackblitz above
2.  Click on id filter
3.  Enter any value
4.  See no filtering happening

## Expected behavior

Rows filtered according to filter input

## Versions

**App**

- Angular: 8
- Clarity: 2.4.2

## Additional notes

Same breakage visible in the official docs: https://clarity.design/documentation/datagrid/binding-properties
The filter on the 'wins' column is broken

**Comment by bdryanovski** [comment]
I think that this is a regression on an old issue - I can't find now the issue but will submit a PR with a fix..

**Comment by gnomeontherun** [comment]
@bdryanovski did you backport this to v2 & v3? If so please close this out.

**Comment by cagolh** [comment]
It looks like this is working for v2 but not v3. We're hoping to upgrade to v3 once it is fixed. Is there any progress on it?

**Comment by mathisscott** [comment]
@bdryanovski 
Has this been backported to v3?

**Comment by bdryanovski** [comment]
From what I'm seeing these changes are already part of v3. 

<img width="584" alt="Screenshot 2020-08-17 at 7 38 50 PM" src="https://user-images.githubusercontent.com/204564/90420924-505fff80-e0c1-11ea-8449-d7e8d0ec6698.png">

If there is some issue with that on v3 that could be something else. I gonna check it now, but the commit is part of the v3 log.

I gonna need some time to verify this.

**Comment by mathisscott** [comment]
Cool. Thanks. Just checking. If it has been fixed and merged, mind closing it?

**Comment by bdryanovski** [comment]
It's an open issue, I manage to reproduce it. 

I don't have a quick solution on top of my head, I may have to rework some of the code here and there to fix it.

For now the workaround for this issue is to change the order of the props: 

```html
<!-- from this -->
<clr-dg-column  clrDgField="id" clrDgColType="number">User ID</clr-dg-column> 

<!-- to this -->
<clr-dg-column clrDgColType="number" clrDgField="id" >User ID</clr-dg-column>
```

The issue comes from when and how the `@Input`s are assign. The `clDgField` depends on the value of `clrDgColType` to work. 

The strange thing is that this was and works for v2 but not for v3, from this, I could assume that this is a valid issue for v4 (did not test it) From the history I'm seeing that we made some changes on how we set and assign the properties for v3 release and Angular9 so this could be a regression from there.

Gonna try to provide a fix for the next release, for now, the workaround is making it non-blocking issue.
